django/db/models/indexes.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# Max length restriction on names imposed by Oracle
+MAX_NAME_LENGTH = 30
```

Comment:
Since we want `name` to be a part of the deconstruction of the index classes, it cannot be different for different backends (because it would be a part of the migrations). For that to happen, we need to decide on a length for the indexes. Since Oracle db already imposes a limit of 30 chars for index names (and 30 is already very small for this purpose) I kept the max-length of the names to 30. I guess the comment `# Max length restriction on names imposed by Oracle` is a bit misleading; I'll change/remove it if that's the case.